packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -865,31 +944,49 @@ void main() {
       ),
     ));
 
-    const String itemLabel = 'Item 2';
-    // Open the menu
+    // Open the menu and highlight the first item.
     await tester.tap(find.byType(DropdownMenu<TestMenu>));
+    await tester.pump();
+    await tester.sendKeyEvent(LogicalKeyboardKey.arrowDown);
     await tester.pumpAndSettle();
-    // Highlight the third item by exact search.
-    await tester.enterText(find.byType(TextField).first, itemLabel);
+    // Search for the last item.
+    await tester.enterText(find.byType(TextField).first, menuChildren.last.label);
     await tester.pumpAndSettle();
-    Finder button2Material = find.descendant(
-      of: find.widgetWithText(MenuItemButton, itemLabel).last,
+    final Finder buttonMaterial = find.descendant(
+      of: find.widgetWithText(MenuItemButton, menuChildren.last.label).last,
       matching: find.byType(Material),
     );
-    Material item2material = tester.widget<Material>(button2Material);
-    expect(item2material.color, themeData.colorScheme.onSurface.withOpacity(0.12));
+    final Material itemMaterial = tester.widget<Material>(buttonMaterial);
+    expect(itemMaterial.color, themeData.colorScheme.onSurface.withOpacity(0.12)); // Menu 1 button is highlighted.
+  }, variant: TargetPlatformVariant.desktop());
 
-    // Search something that matches multiple items.
-    await tester.enterText(find.byType(TextField).first, 'Item');
+  // Regression test for https://github.com/flutter/flutter/issues/151878.
+  testWidgets('Filtering does not cause crush with existing currentHighlight', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: DropdownMenu<TestMenu>(
+          enableFilter: true,
+          requestFocusOnTap: true,
+          dropdownMenuEntries: menuChildren,
+        ),
+      ),
+    ));
+
+    // Open the menu and highlight the third item.
+    await tester.tap(find.byType(DropdownMenu<TestMenu>));
     await tester.pumpAndSettle();
-    // The third item should still be highlighted.
-    button2Material = find.descendant(
-      of: find.widgetWithText(MenuItemButton, itemLabel).last,
-      matching: find.byType(Material),
-    );
-    item2material = tester.widget<Material>(button2Material);
-    expect(item2material.color, themeData.colorScheme.onSurface.withOpacity(0.12));
-  }, variant: TargetPlatformVariant.desktop());
+    await tester.sendKeyEvent(LogicalKeyboardKey.arrowDown);
+    await tester.pumpAndSettle();
+    await tester.sendKeyEvent(LogicalKeyboardKey.arrowDown);
+    await tester.pumpAndSettle();
+    await tester.sendKeyEvent(LogicalKeyboardKey.arrowDown);
+    await tester.pumpAndSettle();
```

Comment:
My understanding is that another test in this file was used to create this one, so I think it's probably fine to leave it as-is.

That being said, the [style guide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-pumpandsettle) recommends against using `pumpAndSettle`. If you're able to switch out those calls for something else, that'd be awesome.

Reply:
Looks great now, thanks!